### ISSUE TITLE
bot.position is old

### ISSUE THREAD

**Issue by hot007** [ASPP/pelita#550]
Hopefully I'm misunderstanding how bot.position is meant to work, but it seems that when we query bot.position we effectively see its location at the end of last turn, rather than where it is displayed on the grid right now. 
e.g. if I query bot.position at the start of a move function (in bots that move) then the reported position does not match the displayed position on the grid by one (including in round 0), in the direction of the move it makes that turn.
Is this expected behaviour?

**Comment by Debilski** [comment]
I am also not sure if I understand correctly. (I know that with the old API there was a lot of misunderstanding of this when students wrote their test functions but this reads a little different.)
Could you go more into detail, @hot007 ?

**Comment by hot007** [comment]
I think it probably is the same thing as you've seen previously. The way we found it was we would pick a target cell and calculate shortest path to that cell, and issue a move call one step along that path. However we found a case (near the boundary) where the bot was displayed in, say, (17,8) and looking to move to (16,8), however that errored because bot.position returned (16,8) (ie this is probably a back and forth move), and thus the shorted path from bot.position to the target cell was a null move, but that was not expected because the bot *was* moving.

To check whether we had done something silly in our function or if it was a property of pelita, we tried running again with moving demo bots and checking bot.position vs displayed position on the grid, and found that consistently bot.position returned the "current" position (e.g. in round 0 starting position (1,14)) but the grid displayed the bot in the location of its move (e.g. (1,13)), thus making it appear that bot.position was returning a previous value not a current value. 

But the other way to interpret that is bot.position really *is* the correct value for that round, but the grid is displaying the position after the move has executed, even though in the debugger we have entered the move function but not yet executed the move, as I understand it? So yes, I think you're correct @otizonaizit, that we were misunderstanding what the game GUI was showing.
This wasn't a difficult thing to get around in our code, but we weren't sure if we were not understanding what bot.position is with respect to what's displayed on the game grid, and what "current" really means, is it what we see or what we can query? Sounds like the answer is to trust bot.position.
